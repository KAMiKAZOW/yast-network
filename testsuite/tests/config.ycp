{

include "testsuite.ycp";

map CONFIG = $[];
map READ = $[];

READ = $[
    "network" : $[
	"section" : $[
	    "eth0" : nil,
	    //"eth0:1" : nil,
	    "eth1" : nil,
	    "ppp2" : nil,
	    "ippp3" : nil,
	    "ppp5.bak" : nil,
	    "pcmcia" : nil,
	    // "pcmcia" : nil, -> brakes regexpsub without check
	],
	"value" : $[
	    "eth0" : $[ "IPADDR" : "0"],
	    //"eth0:1" : $[ "IPADDR" : "0:1"],
	    "eth1" : $[ "IPADDR" : "1"],
	    "ppp2" : $[ "IPADDR" : "2"],
	    "ippp3" : $[ "IPADDR" : "3"],
	],
    ],
    "probe" : $[
	"system" : [],
    ],
];

TESTSUITE_INIT([READ], nil);

include "network/config.ycp";

DUMP("ReadConfig");
TEST(``(ReadConfig("")), [READ], nil);
DUMP(sformat("all=%1",CONFIG));
TEST(``(ReadConfig("eth")), [READ], nil);
DUMP(sformat("eth=%1",CONFIG));
TEST(``(ReadConfig("ppp")), [READ], nil);
DUMP(sformat("ppp=%1",CONFIG));
TEST(``(ReadConfig("ippp")), [READ], nil);
DUMP(sformat("ippp=%1",CONFIG));
TEST(``(ReadConfig("tr")), [READ], nil);
DUMP(sformat("tr=%1",CONFIG));
TEST(``(ReadConfig("ppp|eth")), [READ], nil);
DUMP(sformat("ppp|eth=%1",CONFIG));
TEST(``(ReadConfig("eth|tr")), [READ], nil);
DUMP(sformat("eth|tr=%1",CONFIG));

DUMP("LocateDevices");
CONFIG=$[ "eth" : $[ "0" : $[ "BOOTPROTO" : "dhcp" ]]];
TEST(``(LocateDevices("BOOTPROTO", "dhcp")), [], nil);
CONFIG=$[ "eth" : $[ "0" : $[ "BOOTPROTO" : "" ]]];
TEST(``(LocateDevices("BOOTPROTO", "dhcp")), [], nil);
CONFIG=$[ "eth" : $[ "0" : $[ "BOOTPROTO" : "static" ]]];
TEST(``(LocateDevices("BOOTPROTO", "dhcp")), [], nil);
CONFIG=$[ "eth" : $[ "0" : $[ "BOOTPROTO" : "static" ], "1" : $[ "BOOTPROTO" : "dhcp"]]];
TEST(``(LocateDevices("BOOTPROTO", "dhcp")), [], nil);
CONFIG=$[ "eth" : $[ "0" : $[ "BOOTPROTO" : "static" ]], "tr" : $["1" : $[ "BOOTPROTO" : "dhcp"]]];
TEST(``(LocateDevices("BOOTPROTO", "dhcp")), [], nil);

}
