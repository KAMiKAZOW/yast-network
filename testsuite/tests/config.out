Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system []
Dump	ReadConfig
Dir	.network.section: ["eth0", "eth1", "ippp3", "pcmcia", "ppp2", "ppp5.bak"]
Dir	.network.value."eth0": ["IPADDR"]
Read	.network.value."eth0".IPADDR "0"
Dir	.network.value."eth1": ["IPADDR"]
Read	.network.value."eth1".IPADDR "1"
Dir	.network.value."ippp3": ["IPADDR"]
Read	.network.value."ippp3".IPADDR "3"
Dir	.network.value."ppp2": ["IPADDR"]
Read	.network.value."ppp2".IPADDR "2"
Return	true
Dump	all=$["eth":$["0":$["IPADDR":"0"], "1":$["IPADDR":"1"]], "ippp":$["3":$["IPADDR":"3"]], "ppp":$["2":$["IPADDR":"2"]]]
Dir	.network.section: ["eth0", "eth1", "ippp3", "pcmcia", "ppp2", "ppp5.bak"]
Dir	.network.value."eth0": ["IPADDR"]
Read	.network.value."eth0".IPADDR "0"
Dir	.network.value."eth1": ["IPADDR"]
Read	.network.value."eth1".IPADDR "1"
Return	true
Dump	eth=$["eth":$["0":$["IPADDR":"0"], "1":$["IPADDR":"1"]]]
Dir	.network.section: ["eth0", "eth1", "ippp3", "pcmcia", "ppp2", "ppp5.bak"]
Dir	.network.value."ppp2": ["IPADDR"]
Read	.network.value."ppp2".IPADDR "2"
Return	true
Dump	ppp=$["ppp":$["2":$["IPADDR":"2"]]]
Dir	.network.section: ["eth0", "eth1", "ippp3", "pcmcia", "ppp2", "ppp5.bak"]
Dir	.network.value."ippp3": ["IPADDR"]
Read	.network.value."ippp3".IPADDR "3"
Return	true
Dump	ippp=$["ippp":$["3":$["IPADDR":"3"]]]
Dir	.network.section: ["eth0", "eth1", "ippp3", "pcmcia", "ppp2", "ppp5.bak"]
Return	true
Dump	tr=$[]
Dir	.network.section: ["eth0", "eth1", "ippp3", "pcmcia", "ppp2", "ppp5.bak"]
Dir	.network.value."eth0": ["IPADDR"]
Read	.network.value."eth0".IPADDR "0"
Dir	.network.value."eth1": ["IPADDR"]
Read	.network.value."eth1".IPADDR "1"
Dir	.network.value."ppp2": ["IPADDR"]
Read	.network.value."ppp2".IPADDR "2"
Return	true
Dump	ppp|eth=$["eth":$["0":$["IPADDR":"0"], "1":$["IPADDR":"1"]], "ppp":$["2":$["IPADDR":"2"]]]
Dir	.network.section: ["eth0", "eth1", "ippp3", "pcmcia", "ppp2", "ppp5.bak"]
Dir	.network.value."eth0": ["IPADDR"]
Read	.network.value."eth0".IPADDR "0"
Dir	.network.value."eth1": ["IPADDR"]
Read	.network.value."eth1".IPADDR "1"
Return	true
Dump	eth|tr=$["eth":$["0":$["IPADDR":"0"], "1":$["IPADDR":"1"]]]
Dump	LocateDevices
Return	["eth0"]
Return	[]
Return	[]
Return	["eth1"]
Return	["tr1"]
